inference-engine/tests/functional/plugin/cpu/single_layer_tests/extract_image_patches.cpp

```diff
@@ -0,0 +1,140 @@
+        std::tie(basicParamsSet, cpuParams) = obj.param;
+
+        std::ostringstream result;
+        result << LayerTestsDefinitions::ExtractImagePatchesTest::getTestCaseName(testing::TestParamInfo<extractImagePatchesTuple>(
+            basicParamsSet, 0));
+
+        result << CPUTestsBase::getTestCaseName(cpuParams);
+        return result.str();
+    }
+protected:
+    void SetUp() override {
+        extractImagePatchesTuple basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = this->GetParam();
+        std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+
+        std::vector<size_t> inputShape, kernel, strides, rates;
+        ngraph::op::PadType pad_type;
+        InferenceEngine::Precision netPrecision;
+        std::tie(inputShape, kernel, strides, rates, pad_type, netPrecision, inPrc, outPrc, inLayout, targetDevice) = basicParamsSet;
+        selectedType = std::string("unknown_") + netPrecision.name();
+
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+        auto inputNode = std::make_shared<ngraph::opset6::Parameter>(ngPrc, ngraph::Shape(inputShape));
+        ngraph::ParameterVector params = {inputNode};
+
+        auto extImgPatches = std::make_shared<ngraph::opset6::ExtractImagePatches>(
+                inputNode, ngraph::Shape(kernel), ngraph::Strides(strides), ngraph::Shape(rates), pad_type);
+        ngraph::ResultVector results{std::make_shared<ngraph::opset6::Result>(extImgPatches)};
+        function = std::make_shared<ngraph::Function>(results, params, "ExtractImagePatches");
+    }
+};
+
+TEST_P(ExtractImagePatchesLayerCPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    Run();
+    CheckPluginRelatedResults(executableNetwork, "ExtractImagePatches");
+}
+
+namespace {
+    const std::vector<ngraph::op::PadType> autoPads = {ngraph::op::PadType::VALID, ngraph::op::PadType::SAME_UPPER, ngraph::op::PadType::SAME_LOWER};
+    const std::vector<Precision> netPrecision = {Precision::I8, Precision::BF16, Precision::FP32};
+    const CPUSpecificParams CPUParams = emptyCPUSpec;
+
+/* ============= 1D ============= */
+const auto Layer_params_1D = ::testing::Combine(
+        ::testing::Values(std::vector<size_t> {1, 1, 1, 37}),   // InShapes
+        ::testing::Values(std::vector<size_t> {1, 5}),          // Kernel Sizes
+        ::testing::Values(std::vector<size_t> {1, 2}),          // Strides
+        ::testing::Values(std::vector<size_t> {1, 3}),          // Rates
+        ::testing::ValuesIn(autoPads),
+        ::testing::ValuesIn(netPrecision),
+        ::testing::Values(InferenceEngine::Precision::UNSPECIFIED),
+        ::testing::Values(InferenceEngine::Precision::UNSPECIFIED),
+        ::testing::Values(InferenceEngine::Layout::ANY),
+        ::testing::Values(CommonTestUtils::DEVICE_CPU));
+
+INSTANTIATE_TEST_CASE_P(smoke_ExtractImagePatches_CPU_1D, ExtractImagePatchesLayerCPUTest,
```

Comment:
Why we need 4 separate instances here? Seems like all cases should be unified inside one.

Reply:
This was done for two reasons. First, I wanted to check just a few special cases (which are quite illustrative and can be easily analyzed with pen and paper), and avoid unnecessary cross-checks (since plenty of them are already performed in shared tests). Secondly, 1D tests could not be directly combined with the rest of the tests because 2D kSizes (strides or rates) would produce an error with 1D input shapes. 
Nevertheless, I can still squash the all tests for a unified 4D input shape (so I did), if it is better to have only one instance. Despite the introduced cross-checks, the suite is still pretty fast, so I believe we can go with it.